plugins/discovery-ec2/src/main/java/org/opensearch/discovery/ec2/AwsEc2SeedHostsProvider.java

```diff
@@ -119,41 +119,41 @@ public List<TransportAddress> getSeedAddresses(HostsResolver hostsResolver) {
     }
 
     protected List<TransportAddress> fetchDynamicNodes() {
-
         final List<TransportAddress> dynamicHosts = new ArrayList<>();
 
-        final DescribeInstancesResult descInstances;
-        try (AmazonEc2Reference clientReference = awsEc2Service.client()) {
+        final DescribeInstancesResponse descInstances;
+        try (AmazonEc2ClientReference clientReference = awsEc2Service.client()) {
             // Query EC2 API based on AZ, instance state, and tag.
 
             // NOTE: we don't filter by security group during the describe instances request for two reasons:
             // 1. differences in VPCs require different parameters during query (ID vs Name)
             // 2. We want to use two different strategies: (all security groups vs. any security groups)
-            descInstances = SocketAccess.doPrivileged(() -> clientReference.get().describeInstances(buildDescribeInstancesRequest()));
-        } catch (final AmazonClientException e) {
-            logger.info("Exception while retrieving instance list from AWS API: {}", e.getMessage());
-            logger.debug("Full exception:", e);
+            DescribeInstancesRequest instancesRequest = buildDescribeInstancesRequest();
+            descInstances = SocketAccess.doPrivileged(() -> clientReference.get().describeInstances(instancesRequest));
+        } catch (final RuntimeException e) {
+            logger.warn("Exception while retrieving instance list from AWS API: {}", e.getMessage());
```

Comment:
It seem like we just logging the same information twice: once as `e.getMessage()` and second time as `e`. We probably should log once or log at different log levels

Reply:
Combined.